task_2/task_2.py

```diff
@@ -0,0 +1,430 @@
+
+class Var:
+    dir_script = ''
+    temp_name_data_file = 'temp.csv'
+    log_file_name = 'log_file.log'
+    log_file_path = '.\\'
+    current_dir = ''
+    min_decade = 1960
+
+    output_arj_name = 'output_data.zip'
+    output_folder_default = 'output'
+    output_folder = ''
+    output_file_name = 'out_data.csv'
+    filter_gender = 'male'
+    filter_rows = ''
+    log_level = 'DEBUG'
+
+    temp_data_list_dict = []
+
+# logging.error("Sam log:Error msg")
+# logging.debug("Debug message")
+
+
+# 1.  Set up new file logger and don’t forget to log information about each step
+def create_logger(work_path, log_file_name, log_level):
+    # [DEBUG, INFO, WARNING, ERROR, CRITICAL]
+    print(f'Path\nwork_path = {work_path}\nlog_file_name = {log_file_name}\nlog_level = {log_level}')
+    if not os.path.exists(work_path):
+        os.makedirs(work_path)
+    path = os.path.join(work_path, log_file_name)
+    logging.basicConfig(filename=path,
+                        # level=logging.WARNING,
+                        # level=logging.DEBUG,
+                        # level=logging.INFO,
+                        # level=logging.ERROR,
+                        # level=logging.CRITICAL,
+                        level=log_level,
+                        format='%(asctime)s:%(levelname)s:%(name)s:%(message)s')
+    logger = logging.getLogger("Def_Create_logger")
+    # logger.setLevel(logging.DEBUG)
+    # logger.setLevel(log_level)
+    logging.info('-------- Start script ---------')
+    logger.debug('Logger start')
+
+
+# 2.  Explore the data provider site (https://randomuser.me/documentation) and
+#     download a csv file with 5k user accounts using your script.
+def get_random_users_list(url, file_out):
+    """
+    Call the data provider site https://randomuser.me/documentation and
+    download a csv file with 5k user accounts using your script.
+    :param url:
+    :param file_out:
+    :return: users dict list
+    """
+    logger = logging.getLogger('get_user()')
+    logger.debug(f'requests url:{url}\nresult in file_out{file_out}')
+    try:
+        response = requests.get(url=url)
+        response.raise_for_status()
+    except requests.exceptions.RequestException as e:
+        logger.error("An error occurred: %s", e)
+        print(f"An error occurred: {e}")
+    #     break
+    else:
+        # users_dict_list = []
+        count_users = response.text.count('\n')
+
+        logger.info('Request data is OK')
+        logger.info('Request url:' + response.url)
+        logger.info(f'Downloaded {count_users} accounts of users.')
+
+        with open(file_out, "w", newline='', encoding="utf-8") as file:
+            file.write(response.text)
+            logger.info(f'Users data save in file {file_out}')
+
+        # users_dict_list = [csv.DictReader(response.text.splitlines()[1:], fieldnames=response.text.splitlines()[0].split(','))]
+        # logger.debug('Create dict to users - users_dict_list')
+
+        return count_users
```

Comment:
look, you have only two lines of functionality here 86-87. But the code looks huge. I would recommend you delete the comments, if you don't need them. or uncomment, if you need. also you could decrease amount of logging. that's too much as for me. Stick to the rule: **code should be as simple and readable as possible.**